mailpoet/lib/Automation/Engine/Control/StepHandler.php

```diff
@@ -163,4 +176,24 @@ private function handleStep(array $args): void {
     $this->actionScheduler->enqueue(Hooks::WORKFLOW_STEP, $nextStepArgs);
     // TODO: allow long-running steps (that are not done here yet)
   }
+
+  /** @return SubjectEntry<Subject<Payload>>[] */
+  private function getSubjectEntries(WorkflowRun $workflowRun, array $requiredSubjectKeys): array {
+    $subjectDataMap = [];
+    foreach ($workflowRun->getSubjects() as $data) {
+      $subjectDataMap[$data->getKey()] = array_merge($subjectDataMap[$data->getKey()], [$data]);
```

Comment:
`$subjectDataMap[$data->getKey()] might not exist, maybe better:

`$subjectDataMap[$data->getKey()] = array_merge($subjectDataMap[$data->getKey()]??[], [$data]);`

Reply:
Good catch, fixed.